### PR TITLE
feat: compatible with postgres interval type

### PR DESCRIPTION
**Pull request by QuenKar**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

_PLEASE DO NOT LEAVE THIS EMPTY !!!_

Please explain IN DETAIL what the changes are in this PR and why they are needed:

implement `common_time::interval` convert to/from `postgres interval` using `PgInterval`

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
https://github.com/GreptimeTeam/greptimedb/pull/1952#discussion_r1272953575
Closes: https://github.com/GreptimeTeam/greptimedb/issues/2098

### CODE REVIEW

**Review by killme2008 — APPROVED** [review]
LGTM

**Comment by killme2008** [comment]
@sunng87 Please take a look.

File: src/servers/src/postgres/types.rs [hunk 1]
```diff
@@ -524,6 +549,113 @@ pub(super) fn param_types_to_pg_types(
     Ok(types)
 }
 
+#[derive(Debug, Clone, Copy, Default)]
+pub struct PgInterval {
```
Comment (sunng87) [inline src/servers/src/postgres/types.rs:553]: Can we put this struct to a dedicated module say `types/interval.rs`? I'm afraid this module is going to be too large when more types added.

Comment (QuenKar) [reply src/servers/src/postgres/types.rs:553]: thx, I will refactor it. ๐•ᴗ•๐

**Review by sunng87 — APPROVED** [review]

**Comment by sunng87** [comment]
@QuenKar Thank you!

**Pull request merged**